src/charm.py

```
@@ -132,9 +127,9 @@ def _on_update_status(self, _: EventBase) -> None:  # noqa: C901
             self.model.unit.status = BlockedStatus("Cannot relate to more than one grafana-agent")
             return
 
-        config_valied, confg_valid_message = self.validate_exporter_configs()
-        if not config_valied:
-            self.model.unit.status = BlockedStatus(confg_valid_message)
+        config_valid, config_valid_message = self.validate_exporter_configs()
```

> This PR aims to fix this by removing the unnecessary redfish credential validation step during the install event since it will be performed as part of both the update status and config changed events. This ensures that the exporter is installed before invoking the restart_exporter method.

I understand that if you don't have valid exporter configs on the installation (most likely), it will end up not properly installing the exporter and this can have a weird state, so I'm totally favorable of removing this.

However, another thing showed up to me while  reading the `_on_install_or_upgrade`. If it fails to install the charm is setting to `blocked state`, and this does not look right to me. Blocked means that the user should manually do something. What the user should do if the installation fails? IMO, we should raise an exception and let to be on error state.

---

To be more precise, we should remove any statement about the redfish credential validation since it's a bug before having redfish credential validation step (for example, you can deploy with a wrong `exporter-log-level`, the same problem should arise); that step only reveals the problem now.

> However, another thing showed up to me while reading the _on_install_or_upgrade. If it fails to install the charm is setting to blocked state,...

+1, it does not seem to me that we cannot do anything about install hook fail, the charm should be in error state, and it's bug that we should fix. But that should be raised in another issue

---

though, I agree that the issue and PR description / title did not capture the root cause

---

@gabrielcocenza - Appreciate the suggestion for the PR description. I've made changes accordingly.
@gabrielcocenza , @chanchiwai-ray - I also agree about the exporter install failure bug. I've raised a separate issue to track this (#203)

---

I suggest to move the discussion to somewhere else. There are some gray area between error/blocked status. You can also find similiar discussion in discourse. 
https://discourse.charmhub.io/t/its-probably-ok-for-a-unit-to-go-into-error-state/13022/24

> However, another thing showed up to me while reading the _on_install_or_upgrade. If it fails to install the charm is setting to blocked state, and this does not look right to me. Blocked means that the user should manually do something. What the user should do if the installation fails? IMO, we should raise an exception and let to be on error state.

https://juju.is/docs/sdk/styleguide#heading--error-handling
If installation fails due to network connectivity issues, the operator is responsible for addressing the network problem, as it's not a bug within the charm itself. Therefore, setting the status to "blocked" is reasonable in this scenario. In summary, any error state indicates a potential bug. 

Since Ashley changed the title, please @gabrielcocenza close this discussion thread. thanks!

---

Resolving this conversation since a separate issue has been created for it and the PR desc has been updated.